### DISCUSSION TITLE
Add OpenSUSE package

### DISCUSSION THREAD

**Discussion by sith-on-mars** [mullvad/mullvadvpn-app#8915] [Feature requests & Ideas]
It would be nice that if there is a official OpenSUSE version. OpenSUSE is a a well-known, very established Linux distro and has been around since 2005. Its Enterprise edition, SUSE, existed since 2000. OpenSUSE uses RPM package so it's pretty similar to Fedora. However, during the installation on my OpenSUSE Tumbleweed using the Fedora package, I encountered two issues:

1. The installer says dependency `dbus-libs` cannot be found. This warning can be ignored since OpenSUSE already installs this dependency actually by default, but just with a different naming convention.
2. Startup on boot doesn't work and requires manual fixing. One needs to edit `mullvad-daemon.service` with this command input: `sudo nano /opt/Mullvad\ VPN/resources/mullvad-daemon.service`, then change `Wants=network.target` to `Wants=network-online.target`. You can see the detailed discussion [here](https://forums.opensuse.org/showthread.php/537394-Can-t-get-service-to-start-on-boot).

Having an officially supported OpenSUSE package should be a quick job, since the OpenSUSE package is pretty similar to Fedora package with a few changes.

**Comment by faern** [comment]
Creating a single OpenSUSE compatible RPM is probably a quick job. But it's also going to be outdated quickly. What's harder is to automate the creation of this for every build. The extra build time needed. More artifacts to keep track of and distribute. More complex website download page etc.

Yes, we want to reach out via more distribution channels and in more formats. But each way is a commitment that will cost time and resources forever, not a one time investment.

Thanks for the feedback and the detailed steps needed to produce an OpenSUSE compatible installer! Hopefully we'll get there eventually.

**Comment by xadaemon** [comment]
~~I managed to get it working with relative ease by symlinking from `/opt/Mullvad VPN` to `/opt/mullvadVPN`, since systemd was complaining about not finding the file.
Bellow is how my file ended up:~~
```
# Systemd service unit file for the Mullvad VPN daemon

[Unit]
Description=Mullvad VPN daemon
Wants=network-online.target
After=network-online.target
After=NetworkManager.service
After=systemd-resolved.service
StartLimitBurst=5
StartLimitIntervalSec=20

[Service]
Restart=always
RestartSec=1
ExecStart=/opt/MullvadVPN/resources/mullvad-daemon -v --disable-stdout-timestamps

[Install]
WantedBy=multi-user.target
```
~~edit: clarify why the symlink.~~
# UPDATE:
The issue was something else I found out later but forgotten to update here. Read my latest comment for context.

**Comment by dinotheextinct** [comment]
I would also strongly support this, I am paying for this service and want to use wireguard. In order to use this properly on Opensuse a proper package would be necessary.

**Comment by ghost** [comment]
Just jumping in here to suggest a couple of things which may be of help.

The first is the possibility of an AppImage, which is already an open issue (https://github.com/mullvad/mullvadvpn-app/issues/1123), but it's a bit old and I don't know if it was ever considered.

It might be incredibly useful as it'll work across every distribution and you wouldn't actually need to build separate packages for each release. The downside is file size, since depencies, runtimes, etc. are part of the AppImage itself. The benefit, though, would be one identical file for every Linux user to download, regardless of distro.

I'm unaware of how it works with network interfaces, though, so it might not be option.

The second option is OpenSuse's OBS. It'll also help with automating and maintaining .deb and .rpm files since it's not actually limited to OpenSuse itself and works with packages used on various distributions.

See the developer section on the main site here: https://openbuildservice.org/

Admittedly, I'm not a developer, but I've occasionally found other projects on GitHub which use it for automating packages and updates.

Additionally, because it's technically a repository, I believe you could also allow people to add it as a third party repo which would allow them to update the app using their system's package manager without having to go to the website or the releases page here to download a new updated binary, though that's of course another matter altogether.

Just a couple of small suggestions and I admit I'm unaware of the inner workings of the systems I brought up, but maybe it could be of help.

**Comment by TruckerZer0** [comment]
+1 On having an openSUSE OBS Mullvad repo, something like "home:mullvad:mullvadvpn-app",  even better if cross-distro; 

Or what about simply removing "dbus-libs" from the rpm package installation requisites?

**Comment by myrust** [comment]
> Creating a single OpenSUSE compatible RPM is probably a quick job. But it's also going to be outdated quickly. What's harder is to automate the creation of this for every build. The extra build time needed. More artifacts to keep track of and distribute. More complex website download page etc.
> 
> Yes, we want to reach out via more distribution channels and in more formats. But each way is a commitment that will cost time and resources forever, not a one time investment.
> 
> Thanks for the feedback and the detailed steps needed to produce an OpenSUSE compatible installer! Hopefully we'll get there eventually.

Use OBS to automate the build and provide packages from the same source for Fedora, Debian, Ubuntu, SUSE Linux Enterprise and other distributions!
https://build.opensuse.org/

**Comment by groenator** [comment]
The problem with the package is a naming issue. 

Here is what zypper is saying:

```bash
sudo zypper --no-gpg-checks in MullvadVPN-2021.4_x86_64.rpm 
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: nothing provides 'dbus-libs' needed by the to be installed mullvad-vpn-2021.4.0-1.x86_64
 Solution 1: do not install mullvad-vpn-2021.4.0-1.x86_64
 Solution 2: break mullvad-vpn-2021.4.0-1.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/c/d/?] (c): c

```

lib-dbus is installed_

```
LANG=C zypper se -si libdbus
Loading repository data...
Reading installed packages...

S  | Name               | Type    | Version                  | Arch   | Repository
---+--------------------+---------+--------------------------+--------+----------------------
i+ | libdbus-1-3        | package | 1.12.20-5.4              | x86_64 | Main Repository (OSS)
i+ | libdbus-1-3-32bit  | package | 1.12.20-5.4              | x86_64 | Main Repository (OSS)
i  | libdbusmenu-glib4  | package | 16.04.0-7.10             | x86_64 | Main Repository (OSS)
i  | libdbusmenu-gtk3-4 | package | 16.04.0-7.9              | x86_64 | Main Repository (OSS)
i+ | libdbusmenu-qt5-2  | package | 0.9.3+16.04.20160218-1.5 | x86_64 | Main Repository (OSS)
```

checking the rpm is showing is a naming scheme error. 

```bash
rpm -q --requires mullvad-vpn 
/bin/sh
/bin/sh
/bin/sh
/bin/sh
/bin/sh
dbus-libs
libXScrnSaver
libnotify
libnsl
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1

```

So maybe the devs can fix this issue easier, maybe create a new RPM for openSuse?

**Comment by faern** [comment]
We are not going to have separate artifacts for OpenSUSE for now. There are more issues than this on OpenSUSE with our app already. Making the installer work better will just encourage more people to try to use it and they will hit the other issues. We don't support Suse and currently don't have the bandwidth to do so sadly. But this issue remains open for some future date when we might have the bandwidth to support this distro.

**Comment by groenator** [comment]
Thank you for your response.

Since is Linux there are other ways how to use mullvad. That's what is so
good about Linux :).

It's good that in future are options to review this PR.

Regards,

**Comment by t-wilis** [comment]
Late on this I know, but why is there a space in the folder name for Mullvad in /opt/'Mullvad VPN'?
Why not just MullvadVPN, Mullvad_VPN, or Mullvad-VPN?

Historically have had issues on various OS's when spaces are introduced in paths.

Just to be clear, I am not requesting you to support OpenSUSE or SUSE. I  know manpower is limited and you do what you can.

**Comment by faern** [comment]
Systems that can't handle spaces have bugs in them. Yes, the pragmatic approach is to avoid spaces to avoid the issues. The correct solution is for those systems to fix their bugs. We started having a space in the name because `electron-builder` that we use to package our app uses the app name as the directory name by default and it was not configurable. So we just went with it. And to change it now would be a bit complicated and not really worth it given that it works fine on the vast majority of all systems/distros.

**Comment by EwolBash** [comment]
> We are not going to have separate artifacts for OpenSUSE for now. There are more issues than this on OpenSUSE with our app already. Making the installer work better will just encourage more people to try to use it and they will hit the other issues. We don't support Suse and currently don't have the bandwidth to do so sadly. But this issue remains open for some future date when we might have the bandwidth to support this distro.

What kind of issues? Is it something we as users could approach Suse about?

**Comment by who-biz** [comment]
For anyone in need of an AppImage compiling source, I have made one available here: https://github.com/who-biz/mullvadvpn-app-appimage

Please see README for additional info.

**Comment by ewal** [comment]
I recently did a fresh install of openSUSE Tumbleweed and got very surprised that the gui app just worked without me doing anything so maybe this problem is fixed. :partying_face: 
Previously I started the daemon manually but those days are hopefully over :slightly_smiling_face:

**Comment by cornfeedhobo** [comment]
@faern can you point me to the spec file that currently builds the fedora rpm?

**Comment by raksooo** [comment]
@cornfeedhobo To build the rpm we use Electron builder, which generates the spec file from the information specified here: https://github.com/mullvad/mullvadvpn-app/blob/main/gui/tasks/distribution.js.

**Comment by cornfeedhobo** [comment]
@raksooo Here is my triage:

I think the simplest solution is to copy the pattern of [the deb config to not include any dependency definitions](https://github.com/mullvad/mullvadvpn-app/blob/880bdb9875be41c4be2196328ebc45a08b51da8b/gui/tasks/distribution.js#L180).

* As reported by others, the app currently works out of the box for most suse users, but they are going to bump into issues with package names not matching those of Fedora.

* I believe the documentation on how to build is sufficient for any distros that want to take up building a native package through their build systems and ensure dependencies align.

* Similarly, any user that's stubborn enough to go searching for your RPM is probably going to be able to navigate installing any missing dependencies. The only cases I anticipate will be related to the system tray icon support. The other dependencies are widely required and will likely already be installed.

* Long term, for the purposes of offloading the maintenance overhead of building these packages and giving your company greater exposure, thought should be given to writing instructions for offline compilation for distros that may want to include this as a package in a "main" repo. Most build environments (e.g. [open build service](https://build.opensuse.org/)) are not allowed to access remote resources during compilation.

If you agree with my assessment, making a PR is trivial.

**Comment by cornfeedhobo** [comment]
@raksooo PR submitted. Please review.

**Comment by cornfeedhobo** [comment]
I could use some other voices on my pr. I firmly believe it's being gated arbitrarily. I've dealt with this in the past with some fedora zealots. Not really sure where we can go, because this is a hard blocker unless they decide to use a different build tool chain.

**Comment by secretmango** [comment]
> Creating a single OpenSUSE compatible RPM is probably a quick job. But it's also going to be outdated quickly. What's harder is to automate the creation of this for every build. The extra build time needed. More artifacts to keep track of and distribute. More complex website download page etc.
> 
> Yes, we want to reach out via more distribution channels and in more formats. But each way is a commitment that will cost time and resources forever, not a one time investment.
> 
> Thanks for the feedback and the detailed steps needed to produce an OpenSUSE compatible installer! Hopefully we'll get there eventually.

so... Flatpak?

**Comment by secretmango** [comment]
> I could use some other voices on my pr. I firmly believe it's being gated arbitrarily. I've dealt with this in the past with some fedora zealots. Not really sure where we can go, because this is a hard blocker unless they decide to use a different build tool chain.

you did some more changes and didn't answer why they are there. This alone blocks merging.

**Comment by dinotheextinct** [comment]
Any news on this?

**Comment by Andriamanitra** [comment]
I would appreciate an official package but as a workaround I was able to get the app working on OpenSUSE Tumbleweed by following these [instructions from reddit](https://www.reddit.com/r/openSUSE/comments/12js02n/any_guides_on_how_to_install_mullvad_vpn_in_tw/k74mlde/):

TL;DR:
1. Download the RPM for Fedora
1. `sudo zypper install rpmrebuild`
1. Modify the RPM with `sudo rpmrebuild -enp MullvadVPN-2023.6_x86_64.rpm` (opens `$EDITOR`)
1. Change 
    * "Requires: dbus-libs" to "Requires: libdbus-1-3"
    * "Requires: libnotify" to "Requires: libnotify4"
1. The modified rpm is saved to `/usr/src/packages/RPMS/x86_64/`. Install it using `sudo zypper install /usr/src/packages/RPMS/x86_64/mullvad-vpn-2023.6-1.x86_64.rpm` (you will get a warning about unsigned package but it can be ignored)

**Comment by cornfeedhobo** [comment]
@firefoxlover I have limited time and I ran out of interest to fight for something that feels painfully obvious. You are welcome to pick up where I left off :)

**Comment by I-I-IT** [comment]
@Andriamanitra your technique worked. I had previously downloaded a very old version (from[ OpenSUSE repo](https://software.opensuse.org/download.html?project=home%3Apawrsley&package=mullvad-vpn)). Following your techniques, I modified the RPM file and then tried to open it, however the installer had an error. But using `sudo zypper install mullvad-vpn-2024.1~beta2-1.x86_64.rpm` succeeded an updated the app, albeit with the unsigned warning.

@Mullvad this would take 2 seconds for you to modify.

**Comment by tomas-kuchta** [comment]
This is how I solved it in openSuSE 15.5:
Download .deb package (not rpm)
sudo zypper install dpkg
sudo dpkg -i MullvadVPN-2024.1_amd64.deb

Just a comment: Given that Ubuntu/Debian artifacts are the same as openSuSE's, both distros use the systemd and networkManager - I am not sure what to think about the extra effort in creating openSuSE specific rpm.

**Comment by secretmango** [comment]
crazy that this works, but this will likely not work on the immutable variants. Creating an opensuse rpm should be a very small change

**Comment by bernado82** [comment]
@Andriamanitra
I hadn't thought of modifying the Mullvad VPN specfile to rename the libnotify package name to libnotify4 because it worked without doing that. But now, I think I'll do it for my next rebuilds.

Few community packages for Mullvad VPN are available in openSUSE repositories:
https://software.opensuse.org/package/mullvadvpn

If after a Tumbleweed/Leap system update, Mullvad VPN display a scrambled window or none worldmap, try to:

- Clear GPU cache with: rm -r ~/.config/Mullvad\ VPN/GPUCache/
- Add option "--disable-gpu" in launcher command (example: /opt/Mullvad\ VPN/mullvad-vpn **--disable-gpu**)
- Re-install same RPM package or use a new version.

**Comment by I-I-IT** [comment]
When I downloaded the mentionned package a few weeks ago, it was severely outdated, so much that there was a warning in the app.

**Comment by vyvir** [comment]
+1 for OpenSUSE support!

**Comment by xuzhao9** [comment]
I have a self-packaged verstion at https://build.opensuse.org/package/show/home:nuklly/mullvadvpn, happy to see if someone finds it is useful.

**Comment by ghost** [comment]
> I would appreciate an official package but as a workaround I was able to get the app working on OpenSUSE Tumbleweed by following these [instructions from reddit](https://www.reddit.com/r/openSUSE/comments/12js02n/any_guides_on_how_to_install_mullvad_vpn_in_tw/k74mlde/):
> 
> TL;DR:
> 
>     1. Download the RPM for Fedora
> 
>     2. `sudo zypper install rpmrebuild`
> 
>     3. Modify the RPM with `sudo rpmrebuild -enp MullvadVPN-2023.6_x86_64.rpm` (opens `$EDITOR`)
> 
>     4. Change
>        
>        * "Requires: dbus-libs" to "Requires: libdbus-1-3"
>        * "Requires: libnotify" to "Requires: libnotify4"
> 
>     5. The modified rpm is saved to `/usr/src/packages/RPMS/x86_64/`. Install it using `sudo zypper install /usr/src/packages/RPMS/x86_64/mullvad-vpn-2023.6-1.x86_64.rpm` (you will get a warning about unsigned package but it can be ignored)

Surely just doing `sudo zypper install MullvadVPN-2025.5._x86_64.rpm` is enough as long you ignore the signature verification

**Comment by bernado82** [comment]
On my Tumbleweed (build 20250418), MullvadVPN 2025.5 can be installed directly without rebuilding from Fedora original package. But NOT on my Leap 15.6, that's why you must made a rebuild only for Leap with RPMRebuild utility.

If you install Mullvad VPN with original (Fedora) RPM, you can import the GPG
signing key with the command:

 `sudo rpm --import https://mullvad.net/media/mullvad-code-signing.asc`

or

`wget https://mullvad.net/media/mullvad-code-signing.asc`

_(and by importing manually the GPG key file mullvad-code-signing.asc with YaST repositories manager)_

   
After importing official signing key, you don't get anymore signature error when installing new version.

**Comment by nico198x** [comment]
yes please!

particular attention could be spent on SELinux compatibility and updating packaging practices, for example not writing runtime files to /usr instead of a place like /run or /var.

i very much want to use Mullvad on Immutable systems such as Aeon or Kalpa Desktops, but need some help from the Mullvad devs to make that a reality!

**Comment by bernado82** [comment]
On openSUSE **Tumbleweed**, Mullvad VPN 2025.x can be installed directly _"out-of-the-box"_ from Fedora RPM package (but **NOT on Leap 15.6**, rebuilt still required!).

If you want install MullvadVPN from original Fedora RPM, you can import official **GPG signing key** with the command:

`sudo rpm --import https://mullvad.net/media/mullvad-code-signing.asc`

or `wget https://mullvad.net/media/mullvad-code-signing.asc` (and by importing manually the GPG key file **mullvad-code-signing.asc** with **YaST repositories manager**)

If you want to get **automatic updates with Zypper** & Discover, you've just to add official Fedora repository with the command (or manually with **YaST**):

`sudo zypper ar -f https://repository.mullvad.net/rpm/stable/x86_64/ MullvadVPN`

and this one if you want install beta versions also:

`sudo zypper ar -f https://repository.mullvad.net/rpm/beta/x86_64/ MullvadVPN_beta`

![Image](https://github.com/user-attachments/assets/2484063b-f620-4541-9326-2e2980b01316)
